app/controllers/observations_controller/show.rb

```diff
@@ -117,4 +118,12 @@ def external_sites_user_can_add_links_to(obs)
       @user.external_sites.where.not(id: obs_site_ids)
     end
   end
+
+  # Incurs a costly namings lookup if called in the partial outside show_obs
+  # e.g., in the lightbox caption
+  def register_namings_for_textile_in_notes
+    return unless @observation.notes?
+
+    Textile.register_name(*@observation.namings.map(&:name))
+  end
 end
```

Comment:
This avoids the costly extra `naming` includes on the indexes where the captions will appear (in the lightbox).
It only registers the `Textile` naming on the show_obs page.